src/Aperture/elliptical.jl

```diff
@@ -0,0 +1,100 @@
+export EllipticalAperture,
+       EllipticalAnnulus
+
+"""
+   EllipticalAperture(x, y, a, b, theta)
+
+An elliptical aperture.
+
+Where a >= b > 0 and theta in degrees
+
+# Examples
+```jldoctest
+julia> EllipticalAperture(2,2,2,2,1.2)
+EllipticalAperture(2.0, 2.0, a=2.0, b=2.0, theta=1.2)
+
+```
+"""
+struct EllipticalAperture{T <: Number} <: AbstractAperture
+    x::T
+    y::T
+    a::T
+    b::T
+    theta::T
+end
+
+function EllipticalAperture(x, y, a, b, theta)
+    if (b < 0.0 || a < b )
+            error("Invalid ellipse parameters. Require a >= b > 0.0")
+    end
+    theta = mod(theta, 360)
+    return EllipticalAperture(promote(x, y, a, b, theta)...)
+end
+
+function Base.show(io::IO, e::EllipticalAperture)
+    print(io, "EllipticalAperture($(e.x), $(e.y), a=$(e.a), b=$(e.b), theta=$(e.theta)°)")
+end
+
+function oblique_coefficients(a, b, theta)
+    sintheta, costheta = sincos(deg2rad(theta))
+    a2 = a^2
+    b2 = b^2
+    cxx = costheta^2 / a2 + sintheta^2 / b2
+    cyy = sintheta^2 / a2 + costheta^2 / b2
+    cxy = 2*costheta*sintheta * (1/a2 - 1/b2)
+    return cxx, cyy, cxy
+end
+
+function bbox(e::EllipticalAperture)
+
+    sintheta, costheta = sincos(deg2rad(e.theta))
+
+    t = atan((-e.b*tan(deg2rad(e.theta)))/e.a)
+
+    xmin = e.x + e.a*cos(t)*costheta - e.b*sin(t)*sintheta
+    xmax = xmin
+
+    for n in -2:2
+        xmin = min(xmin, e.x + e.a*cos(t + n*pi)*costheta - e.b*sin(t + n*pi)*sintheta)
+    end
+
+    for n in -2:2
+        xmax = max(xmax, e.x + e.a*cos(t + n*pi)*costheta - e.b*sin(t + n*pi)*sintheta)
+    end
+
+    t = atan((e.b*cot(deg2rad(e.theta)))/e.a)
+
+    ymin = e.y + e.b*sin(t)*costheta + e.a*cos(t)*sintheta
+    ymax = ymin
+
+    for n in -2:2
+        ymin = min(ymin, e.y + e.b*sin(t + n*pi)*costheta + e.a*cos(t + n*pi)*sintheta)
+    end
+
+    for n in -2:2
+        ymax = max(ymax, e.y + e.b*sin(t + n*pi)*costheta + e.a*cos(t + n*pi)*sintheta)
+    end
```

Comment:
In all these loops you can save some time by computing `sin` and `cos` at the same time with `sincos`.

Double check what I'm saying, but I think that for all trigonometric functions used here there is the version with degree input: `tand`, `cotd`, so that you shouldn't directly call `deg2rad` anywhere.

Reply:
TIL `sincosd` exists.

Reply:
I checked the documentation for sincosd, it is supported only in v1.3 and above. What practise should I follow so that this runs amongst different versions of Julia?